sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsRepositoryAsyncClient.java

```diff
@@ -29,27 +36,31 @@
 @ServiceClient(builder = ModelsRepositoryClientBuilder.class, isAsync = true)
 public final class ModelsRepositoryAsyncClient {
     private static final ClientLogger logger = new ClientLogger(ModelsRepositoryAsyncClient.class);
-    private final ObjectMapper mapper;
+    private static final String MODELS_REPOSITORY_TRACING_NAMESPACE_VALUE = "Azure.Iot.ModelsRepository";
     private final ModelsRepositoryServiceVersion serviceVersion;
     private final ModelsRepositoryAPIImpl protocolLayer;
-    private final JsonSerializer serializer;
-    private static final String MODELS_REPOSITORY_TRACING_NAMESPACE_VALUE = "Azure.Iot.ModelsRepository";
+    private final RepositoryHandler repositoryHandler;
+    private final DependencyResolutionOptions defaultDependencyResolutionOption;
```

Comment:
I'm not so sure about having a default value at the class level for this when it is an option at a per-method level. Would it be simpler to just have the user provide a value each time to the method and for our SDK to have a hardcoded default for when it isn't specified?

Reply:
I don't disagree with you there, the decision we came up with during the .NET review, was that we want to take a default value for all the API calls at construction time and have the option to override it for each individual API call. Since there is no client options pattern in java, I figured this is the best place for it to be for us to mimic the behavior.